### DISCUSSION TITLE
How to use proxy with budibase

### DISCUSSION THREAD

**Discussion by rolandwang19** [Budibase/budibase#6917] [General]
I've set tried everything I can find online, and can't seem to get budibase to use the proxy on my linux server. I've set the proxy in every example I could find online, and ran the container with the proxy. I'm able to curl with the proxy on the server and everything works fine. I just can't get the application to call an external API no matter where I've tried to set the proxy.

/etc/systemd/system/docker.service.d/http-proxy.conf
systemctl edit docker.service
/etc/sysconfig/docker
/etc/yum.conf
/etc/environment
~/.docker/config.json

sudo docker run -d -t \
  --env http_proxy="" \
  --env https_proxy=""\
  --env HTTP_PROXY="" \
  --env HTTPS_PROXY=""\
  --name=budibase \
  -p 10000:80 \
  -v /local/path/data:/data \
  --restart unless-stopped \
  budibase/budibase:latest

**Comment by ThomasK33** [comment]
The `GLOBAL_AGENT_HTTP_PROXY` and `GLOBAL_AGENT_HTTPS_PROXY` env variables did the trick for me

**Reply by shogunpurple** [reply]
We should document this - thanks @ThomasK33

**Reply by rolandwang19** [reply]
Did you just include is as part of your docker run command, or did you set in specific config files? When I do it I get a 502 error.

**Reply by ThomasK33** [reply]
> Did you just include is as part of your docker run command, or did you set in specific config files? When I do it I get a 502 error.

Environment variable on the container itself.

**Reply by shogunpurple** [reply]
https://docs.budibase.com/docs/running-budibase-behind-company-proxy

**Reply by ThomasK33** [reply]
> [docs.budibase.com/docs/running-budibase-behind-company-proxy](https://docs.budibase.com/docs/running-budibase-behind-company-proxy)

You might also want to mention the `GLOBAL_AGENT_NO_PROXY` variable.

**Reply by shogunpurple** [reply]
Updated! Thanks @ThomasK33

**Reply by rolandwang19** [reply]
Thanks for the documentation! Any idea why I might be getting a 502 error?

![image](https://user-images.githubusercontent.com/10952866/181238008-ee53c52c-3e5a-4814-a010-27a945c32be2.png)

**Reply by shogunpurple** [reply]
@rolandwang19 what does your run command look like exactly? Looking at the above - is it:

```
sudo docker run -d -t
--env GLOBAL_AGENT_HTTP_PROXY="http://something.net"
--env GLOBAL_AGENT_HTTPS_PROXY="https://something.net"
--name=budibase
-p 10000:80
-v /local/path/data:/data
--restart unless-stopped
budibase/budibase:latest
```

**Reply by rolandwang19** [reply]
Yeah, that's exactly what it looks like. I also tried altering the docker-compose file with the added environment variables and get the same when I use docker-compose

**Reply by shogunpurple** [reply]
maybe a silly question - but you have changed the `-v /local/path/data:/data` to a valid path on your machine?

**Reply by rolandwang19** [reply]
Yeah, no dice. Even tried removing it.

**Reply by rolandwang19** [reply]
Actually, what path should actually be used? I mapped to local path, and I checked the container and see that it exists, but what's the purposed of the volume and what should actually be in the local path data folder?
